cpp/src/arrow/flight/internal.cc

```diff
@@ -196,7 +207,7 @@ Status FromGrpcStatus(const grpc::Status& grpc_status, grpc::ClientContext* ctx)
       return status;
     }
 
-    if (status.detail()) {
+    if (status.detail() && !arrow_status.detail()) {
```

Comment:
This is going to lose information about the underlying error. I'd rather we parse out the extra error detail separately, and then add it to the existing FlightStatusDetail here, instead of completely overwriting it. If there isn't a FlightStatusDetail, then default to INTERNAL.

Reply:
agreed, I believe we can remove this check completely. See most recent patch